tests/Fixer/Phpdoc/NoSuperfluousPhpdocTagsFixerTest.php

```diff
@@ -0,0 +1,482 @@
+}',
+                '<?php
+class Foo {
+    /**
+     * @param mixed $bar
+     *
+     * @return mixed
+     */
+    public function doFoo($bar) {}
+}',
+            ],
+            'multiple_different_types' => [
+                '<?php
+class Foo {
+    /**
+     * @param SubclassOfBar1|SubclassOfBar2 $bar
+     */
+    public function doFoo(Bar $bar) {}
+}',
+            ],
+            'same_typehint_with_different_casing' => [
+                '<?php
+class Foo {
+    /**
+     */
+    public function doFoo(Bar $bar) {}
+}',
+                '<?php
+class Foo {
+    /**
+     * @param bar $bar
+     */
+    public function doFoo(Bar $bar) {}
+}',
+            ],
+            'multiple_arguments' => [
+                '<?php
+class Foo {
+    /**
+     * @param SubclassOfBar1|SubclassOfBar2 $bar
+     */
+    public function doFoo(Bar $bar, Baz $baz = null) {}
+}',
+                '<?php
+class Foo {
+    /**
+     * @param SubclassOfBar1|SubclassOfBar2 $bar
+     * @param Baz|null $baz
+     */
+    public function doFoo(Bar $bar, Baz $baz = null) {}
+}',
+            ],
+            'with_import' => [
+                '<?php
+use Foo\Bar;
+
+/**
+ */
+function foo(Bar $bar) {}',
+                '<?php
+use Foo\Bar;
+
+/**
+ * @param Bar $bar
+ */
+function foo(Bar $bar) {}',
+            ],
+            'with_root_symbols' => [
+                '<?php
+/**
+ */
+function foo(\Foo\Bar $bar) {}',
+                '<?php
+/**
+ * @param \Foo\Bar $bar
+ */
+function foo(\Foo\Bar $bar) {}',
+            ],
+            'with_mix_of_imported_and_fully_qualified_symbols' => [
+                '<?php
+use Foo\Bar;
+use Foo\Baz;
+
+/**
+ */
+function foo(Bar $bar, \Foo\Baz $baz) {}',
+                '<?php
+use Foo\Bar;
+use Foo\Baz;
+
+/**
+ * @param \Foo\Bar $bar
+ * @param Baz $baz
+ */
+function foo(Bar $bar, \Foo\Baz $baz) {}',
+            ],
+            'with_aliased_imported' => [
+                '<?php
+use Foo\Bar as Baz;
+
+/**
+ */
+function foo(Baz $bar) {}',
+                '<?php
+use Foo\Bar as Baz;
+
+/**
+ * @param \Foo\Bar $bar
+ */
+function foo(Baz $bar) {}',
+            ],
+            'with_unmapped_param' => [
+                '<?php
+use Foo\Bar;
+
+/**
+ * @param Bar
+ */
+function foo(Bar $bar) {}',
+            ],
+        ];
+    }
+
+    /**
+     * @param string      $expected
+     * @param null|string $input
+     *
+     * @dataProvider provideFixPhp70Cases
+     * @requires PHP 7.0
+     */
+    public function testFixPhp70($expected, $input = null)
+    {
+        $this->doTest($expected, $input);
+    }
+
+    public function provideFixPhp70Cases()
+    {
+        return [
+            'same_typehint' => [
+                '<?php
+class Foo {
+    /**
+     *
+     */
+    public function doFoo(Bar $bar): Baz {}
+}',
+                '<?php
+class Foo {
+    /**
+     * @param Bar $bar
+     *
+     * @return Baz
+     */
+    public function doFoo(Bar $bar): Baz {}
+}',
```

Comment:
please add case with input:
```
class Foo {
    /**
     * @param Bar $bar
     *
     * @return Baz
     */
    public function doFoo(Bar $bar) {}
}',
```

input is typehinted on language level, but output is not.
will annotation for input will be removed, while annotation for output will remain?


-------------------

```
class Foo {
    /**
     * @param Bar $bar
     * @param Bax|Baz $baxz
     */
    public function doFoo(Bar $bar, $baxz) {}
}',
```

will `bar` annotation be removed, and `baxz` not?

Reply:
Test cases added.